environments/gravity.py

```diff
@@ -0,0 +1,213 @@
+import numpy as np
+
+from environments import Environment, visualize_rollout
+
+
+class NObjectGravity(Environment):
+
+    """N Object Gravity Atraction System
+
+    Equations of movement are:
+
+        m_i*q'' = G * sum_{i!=j} ( m_i*m_j*(q_j - q_i)/ abs(q_j  - q_i)^3 )
+
+    """
+
+    def __init__(self, mass, gravity_cst, orbit_noise=.01, q=None, p=None):
+        """Contructor for spring system
+
+        Args:
+            mass ([float]): Object masses (kg).
+            gravity_cst (float): Constant for the intensity of gravitational field (m^3/kg*s^2)
+            orbit_noise (float, optional): Noise for object orbits when sampling initial conditions
+            q (ndarray, optional): Object generalized positions in 2-D space: Positions (m). Defaults to None
+            p (ndarray, optional): Object generalized momentums in 2-D space : Linear momentums (kg*m/s). Defaults to None
+        Raises:
+            NotImplementedError: If more than 2 objects are considered
+        """
+        self.mass = mass
+        self.colors = ['y', 'r', 'g', 'b', 'c', 'p', 'w']
+        self.n_objects = len(mass)
+        self.gravity_cst = gravity_cst
+        self.orbit_noise = orbit_noise
+        if self.n_objects > 7:
+            raise NotImplementedError(
+                'Gravity interaction for ' + str(self.n_objects) + ' bodies is not implemented.')
+        super().__init__(q=q, p=p)
+
+    def set(self, q, p):
+        """Sets initial conditions for pendulum
+
+        Args:
+            q (ndarray): Object generalized positions in 2-D space: Positions (m)
+            p (ndarray): Object generalized momentums in 2-D space : Linear momentums (kg*m/s)
+
+        Raises:
+            ValueError: If q and p are not in 2-D space or do not refer to all the objects in space
+        """
+        if q is None or p is None:
+            return
+        if q.shape[0] != self.n_objects or p.shape[0] != self.n_objects:
+            raise ValueError(
+                "q and p do not refer to the same number of objects in the system.")
+        if q.shape[-1] != 2 or p.shape[-1] != 2:
+            raise ValueError(
+                "q and p must be in 2-D space: Position and Linear momentum.")
+        self.q = q.copy()
+        self.p = p.copy()
+
+    def _dynamics(self, t, states):
+        """Defines system dynamics
+
+        Args:
+            t (float): Time parameter of the dynamic equations.
+            states ([float]) Phase states at time t
```

Comment:
I'm not sure I understand what `states` is made of. I understand it needs to contain position and momentum information for each body, but in what format?

Reply:
states is a 1-D array that contains the information of the phase state. In a general way, it will be always np.array([q,p]).reshape(-1).

In the N-body environment you can retrieve the phase state array if you reshape it using .reshape(2, self.n_bodies, 2).

The reason why you need the 1-D array is because scipy.solve_ivp requires it.

Reply:
Ok thanks! can you add that to the docstring? 
```suggestion
            states (numpy.ndarray): 1-D array that contains the information of the phase
                state, in the format of np.array([q,p]).reshape(-1).
```